src/ui/components/SecondaryToolbox/ReactDevTools.tsx

```diff
@@ -229,6 +239,54 @@ function createReactDevTools(
   return ReactDevTools;
 }
 
+type ReactDevToolsStateUpdaterFunction = (reactDevTools: ReactDevToolsType | null) => void;
+
+async function createReactDevToolsForBridgeProtocol(
+  stateUpdater: ReactDevToolsStateUpdaterFunction,
+  annotations: Annotation[],
+  currentPoint: ExecutionPoint,
+  enablePicker: (opts: NodePickerOpts) => void,
+  disablePicker: () => void,
+  onShutdown: () => void
+) {
+  // React DevTools (RD) changed its internal data structure slightly in a minor update.
+  // The result is that Replay sessions recorded with older versions of RD don't play well in newer versions.
+  // We can work around this by checking RD's "bridge protocol" version (which we also store)
+  // and loading the appropriate frontend version to match.
+  // For more information see https://github.com/facebook/react/issues/24219
+  const response = await ThreadFront.evaluate({
+    asyncIndex: 0,
+    text: ` __RECORD_REPLAY_REACT_DEVTOOLS_SEND_MESSAGE__("getBridgeProtocol", undefined)`,
+  });
+  const json: any = await response?.returned?.getJSON();
+  const version = json?.data?.version || 0;
+
+  let ReactDevTools = null;
+  if (version >= 2) {
+    const reactDevToolsInlineModule = await import("react-devtools-inline/frontend");
+    ReactDevTools = await createReactDevTools(
```

Comment:
The `createReactDevTools()` call could be deduplicated by moving it out of the if/then/else block